.github/workflows/ci.yml

```diff
@@ -0,0 +1,200 @@
+---
+name: "CI"
+concurrency: # Cancel any existing runs of this workflow for this same PR
+  group: "${{ github.workflow }}-${{ github.ref }}"
+  cancel-in-progress: true
+on: # yamllint disable
+  push:
+    branches:
+      - "master"
+      - "develop"
+    tags:
+      - "v*"
+  pull_request: ~
+jobs:
+  build:
+    runs-on: "ubuntu-20.04"
+    env:
+      PYTHON_VER: "3.7"
+      NETBOX_VER: "v2.9.11"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v2"
+      - name: "Setup environment"
+        uses: "networktocode/gh-action-setup-poetry-environment@v3"
+      - name: "Build Container"
+        run: "poetry run invoke build"
+  black:
+    runs-on: "ubuntu-20.04"
+    env:
+      PYTHON_VER: "3.7"
+      NETBOX_VER: "v2.9.11"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v3"
+      - name: "Setup environment"
+        uses: "networktocode/gh-action-setup-poetry-environment@v3"
+      - name: "Build Container"
+        run: "poetry run invoke build"
+      - name: "Linting: black"
+        run: "poetry run invoke black"
+    needs:
+      - "build"
+  bandit:
+    runs-on: "ubuntu-20.04"
+    env:
+      PYTHON_VER: "3.7"
+      NETBOX_VER: "v2.9.11"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v3"
+      - name: "Setup environment"
+        uses: "networktocode/gh-action-setup-poetry-environment@v3"
+      - name: "Build Container"
+        run: "poetry run invoke build"
+      - name: "Linting: bandit"
+        run: "poetry run invoke bandit"
+    needs:
+      - "build"
+  pydocstyle:
+    runs-on: "ubuntu-20.04"
+    env:
+      PYTHON_VER: "3.7"
+      NETBOX_VER: "v2.9.11"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v3"
+      - name: "Setup environment"
+        uses: "networktocode/gh-action-setup-poetry-environment@v3"
+      - name: "Build Container"
+        run: "poetry run invoke build"
+      - name: "Linting: pydocstyle"
+        run: "poetry run invoke pydocstyle"
+    needs:
+      - "build"
+  pylint:
+    runs-on: "ubuntu-20.04"
+    env:
+      PYTHON_VER: "3.7"
+      NETBOX_VER: "v2.9.11"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v3"
+      - name: "Setup environment"
+        uses: "networktocode/gh-action-setup-poetry-environment@v3"
+      - name: "Build Container"
+        run: "poetry run invoke build"
+      - name: "Linting: Pylint"
+        run: "poetry run invoke pylint"
+    needs:
+      - "black"
+      - "bandit"
+      - "pydocstyle"
+  unittest:
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: ["3.6", "3.7", "3.8"]
+        netbox-version: ["v2.8.9", "v2.9.11", "v2.10.10", "v2.11.10"]
```

Comment:
That seems like a lot of combinations. Can we reduce options, or do we need to test all combinations?

Reply:
I simply build the same tests as defined in Travis.
I would leave refinement for another PR.